diff --git a/dist/utilities/mergeLocalizedData.js b/dist/utilities/mergeLocalizedData.js
index 97f43812e60028a3207a17778d03174ec7361e66..fd414162f45b23b21515f9ac19b47680f2b61e7a 100644
--- a/dist/utilities/mergeLocalizedData.js
+++ b/dist/utilities/mergeLocalizedData.js
@@ -74,7 +74,7 @@ import { fieldAffectsData, fieldShouldBeLocalized, tabHasName } from '../fields/
                             } else if (Array.isArray(newValue)) {
                                 // Non-localized array - still process children for any localized fields
                                 result[field.name] = newValue.map((newItem, index)=>{
-                                    const existingItem = existingValue?.[index] || {};
+                                    const existingItem = (existingValue?.find((item) => item.id && item.id === newItem.id) || existingValue?.[index]) || {};
                                     return mergeLocalizedData({
                                         configBlockReferences,
                                         dataWithLocales: newItem,
@@ -125,7 +125,7 @@ import { fieldAffectsData, fieldShouldBeLocalized, tabHasName } from '../fields/
                                         block = field.blocks.find((b)=>b.slug === newBlockData.blockType);
                                     }
                                     if (block) {
-                                        const blockData = Array.isArray(existingValue) && existingValue[index] ? existingValue[index] : {};
+                                        const blockData = Array.isArray(existingValue) ? (existingValue.find((b) => b.id && b.id === newBlockData.id) || existingValue[index] || {}) : {};
                                         const merged = mergeLocalizedData({
                                             configBlockReferences,
                                             dataWithLocales: newBlockData,
